### PR TITLE
Allow specifying runtime region for AWS resources

### PR DESCRIPTION
**Pull request by Xantier**
Add an optional parameter to `run` methods which allows changing region at runtime if wanted.

### CODE REVIEW

**Review by punkle — APPROVED** [review]

File: plugins/backend/rag-ai-storage-pgvector/migrations/20231216153432_init_vector_embeddings_store.js [hunk 1]
```diff
@@ -16,38 +16,42 @@
...
-    table
-      .text('content')
-      .notNullable()
-      .comment('Actual content of the embedding. Chunks of text/data');
-    table
-      .jsonb('metadata')
-      .notNullable()
-      .comment(
-        'Metadata of the embedding. Information like entityRef etc. that can be used to identify links to other parts of the system.',
+  if (knex.client.config.client === 'pg') {
```
Comment (punkle) [inline plugins/backend/rag-ai-storage-pgvector/migrations/20231216153432_init_vector_embeddings_store.js:19]: This seems unrelated to the changeset?

**Review by punkle — DISMISSED** [review]

**Review by kissmikijr — APPROVED** [review]
